android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java

```diff
@@ -173,17 +174,41 @@ public void sendNotificationScheduledCore(Bundle bundle) {
 
         Log.d(LOG_TAG, String.format("Setting a notification with id %s at time %s",
                 bundle.getString("id"), Long.toString(fireDate)));
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
-            if (allowWhileIdle && Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-                getAlarmManager().setExactAndAllowWhileIdle(AlarmManager.RTC_WAKEUP, fireDate, pendingIntent);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            if (allowWhileIdle) {
+                if (canScheduleExactAlarms()) {
+                    getAlarmManager().setExactAndAllowWhileIdle(AlarmManager.RTC_WAKEUP, fireDate, pendingIntent);
```

Comment:
"Made inexact alarms the default, and made exact alarms only be scheduled if:
- The device is pre Android 13
- SCHEDULE_EXACT_ALARM permission is granted
"

Is the list of conditions supposed to be joined with OR or AND? 
Suppose it is an OR - but probably helps to make this clear in the descirption